### PR TITLE
[PPP-5353]-XSS Findings For Pentaho-platform-plugin-reporting

### PR DESCRIPTION
**Pull request by Rangashivani**
[PPP-5353]-XSS Findings For Pentaho-platform-plugin-reporting

[PPP-5353]: https://hv-eng.atlassian.net/browse/PPP-5353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 3m 56s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-874/1/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 2m 53s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-874/2/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-874/3/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3m 56s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-874/4/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: core/src/main/javascript/reportviewer/reportviewer.js [hunk 1]
```diff
@@ -818,7 +818,7 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
         if(isRunningIFrameInSameOrigin) {
           if (!top.mantle_initialized) {
             this._topMantleOpenTabRegistration = top.mantle_openTab = function(name, title, url) {
-              window.open(url, '_blank');
+              window.open(xssUtil.sanitizeUrl(url), '_blank');
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/reportviewer.js:821]: Should use the new `xssUtil.openWindow(..)` method instead.

File: core/src/main/javascript/reportviewer/reportviewer.js [hunk 2]
```diff
@@ -1189,7 +1189,7 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
                     hideDlgAndPane(registry.byId('feedbackScreen'));
 
                     //Show loading screen
-                    $('#notification-message').html(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
+                    $('#notification-message').html(xssUtil.sanitizeHtml(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages));
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/reportviewer.js:1192]: Should use the `xssUtil.setHtml(..)` method, after support for jQuery objects has been added to `setHtml(..)` and `setHtmlUnsafe(..)`.

File: core/src/main/javascript/reportviewer/reportviewer.js [hunk 3]
```diff
@@ -1232,7 +1232,7 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
                       isPageCountUpdated = true;
                     }
 
-                    $('#notification-message').html(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
+                    $('#notification-message').html(xssUtil.sanitizeHtml(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages));
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/reportviewer.js:1235]: Idem, for use of `xssUtil.setHtml(...)` given a jQuery object.

File: core/src/main/javascript/reportviewer/reportviewer.js [hunk 4]
```diff
@@ -1242,13 +1242,13 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
                 case "QUEUED":
                 case "WORKING":
                   // Although we are hiding the screen latter, update the label anyway;
-                  $('#notification-message').html(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages);
+                  $('#notification-message').html(xssUtil.sanitizeHtml(_Messages.getString('LoadingPage') + " " + mainJobStatus.page + " " + _Messages.getString('Of') + " " + mainJobStatus.totalPages));
```
Comment (dcleao) [inline core/src/main/javascript/reportviewer/reportviewer.js:1245]: Idem, for use of `xssUtil.setHtml(...)` given a jQuery object.

**Review by dcleao — CHANGES_REQUESTED** [review]

File: core/src/main/javascript/reportviewer/reportviewer.js (hunk 2 above)
Comment (dcleao) [reply core/src/main/javascript/reportviewer/reportviewer.js:1192]: See https://github.com/pentaho/pentaho-platform-plugin-reporting/pull/864#discussion_r1831504782.

**Comment by dcleao** [comment]
New version of old PR, having unresolved comments: https://github.com/pentaho/pentaho-platform-plugin-reporting/pull/864.

File: core/src/main/javascript/reportviewer/reportviewer.js (hunk 4 above)
Comment (Rangashivani) [reply core/src/main/javascript/reportviewer/reportviewer.js]: Updated

File: core/src/main/javascript/reportviewer/reportviewer.js (hunk 3 above)
Comment (Rangashivani) [reply core/src/main/javascript/reportviewer/reportviewer.js]: Updated

File: core/src/main/javascript/reportviewer/reportviewer.js (hunk 2 above)
Comment (Rangashivani) [reply core/src/main/javascript/reportviewer/reportviewer.js]: Updated

File: core/src/main/javascript/reportviewer/reportviewer.js (hunk 1 above)
Comment (Rangashivani) [reply core/src/main/javascript/reportviewer/reportviewer.js]: Updated

**Comment by buildguy** [comment]
### :x: Build failed in 4m 29s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-874/5/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-874/6/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Pull request merged**